koku/api/report/test/ocp/test_views.py

```diff
@@ -667,6 +670,100 @@ def string_to_date(dt):
+            "filter[time_scope_units]": "month",
+        }
+        url = url + "?" + urlencode(params, quote_via=quote_plus)
+        response = APIClient().get(url, **self.headers)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        data = response.data
+        this_month_start = self.dh.this_month_start
+        last_month_start = self.dh.last_month_start
+
+        date_delta = relativedelta.relativedelta(months=1)
+        last_month_end = self.dh.this_month_end - date_delta
+
+        self.cost_term = (
+            self.provider_map.cloud_infrastructure_cost_by_project
+            + self.provider_map.markup_cost_by_project
+            + self.provider_map.cost_model_cost
+            + self.provider_map.cost_model_distributed_cost_by_project
+        )
+
+        with tenant_context(self.tenant):
+            current_total = (
+                OCPUsageLineItemDailySummary.objects.filter(
+                    usage_start__gte=this_month_start.date(), namespace=project_name
+                )
+                .annotate(**{"infra_exchange_rate": Value(Decimal(1.0)), "exchange_rate": Value(Decimal(1.0))})
+                .aggregate(total=self.cost_term)
+                .get("total")
+            )
+            current_total = current_total if current_total is not None else 0
+
+            current_totals = (
+                OCPUsageLineItemDailySummary.objects.filter(
+                    usage_start__gte=this_month_start.date(), namespace=project_name
+                )
+                .annotate(
+                    **{
+                        "date": TruncDayString("usage_start"),
+                        "infra_exchange_rate": Value(Decimal(1.0)),
+                        "exchange_rate": Value(Decimal(1.0)),
+                    }
+                )
+                .values(*["date"])
+                .annotate(total=self.cost_term)
+            )
+
+            prev_totals = (
+                OCPUsageLineItemDailySummary.objects.filter(
+                    usage_start__gte=last_month_start.date(), namespace=project_name
+                )
+                .filter(usage_start__lte=last_month_end.date())
+                .annotate(
+                    **{
+                        "date": TruncDayString("usage_start"),
+                        "infra_exchange_rate": Value(Decimal(1.0)),
+                        "exchange_rate": Value(Decimal(1.0)),
+                    }
+                )
+                .values(*["date"])
+                .annotate(total=self.cost_term)
+            )
+
+        current_totals = {total.get("date"): total.get("total") for total in current_totals}
+        prev_totals = {
+            date_to_string(string_to_date(total.get("date")) + date_delta): total.get("total")
+            for total in prev_totals
+            if date_to_string(string_to_date(total.get("date")) + date_delta) in current_totals
+        }
+
+        prev_total = sum(prev_totals.values())
```

Comment:
Just wondering is this line would ever return `None`  or just error out

Reply:
I think the only way that values comes back empty is if there is something wrong with the bakery loader for our dummy data.